core/src/main/java/org/web3j/protocol/http/HttpService.java

```diff
@@ -157,22 +157,20 @@ protected InputStream performIO(String request) throws IOException {
         okhttp3.Request httpRequest =
                 new okhttp3.Request.Builder().url(url).headers(headers).post(requestBody).build();
 
-        try (okhttp3.Response response = httpClient.newCall(httpRequest).execute()) {
```

Comment:
how come the try with resources has been removed here?

Reply:
I think because the try with resource closes the resource after exiting the scope. However, we return an `InputStream` to the `responseBody` of the `response` in the try. This latter gets closed after exiting the scope then the `InputStream` has nothing to read. Not sure anyway if this is it.

Reply:
I guess `okhttp3.Response` should really be closed. Only solution I see is to return a `ByteArrayInputStream` to "disconnect" the two streams completely.

Reply:
Something like this ? (I just added a change)